insights/parsers/lssap.py

```diff
@@ -0,0 +1,106 @@
+"""
+lssap - Command
+===============
+
+This module provides processing for the output of the ``lssap`` command on SAP systems.
+The spec handled by this command inlude::
+
+    "lssap"                     : CommandSpec("/usr/sap/hostctrl/exe/lssap")
+
+Class ``Lssap`` parses the output of the ``lssap`` command.  Sample
+output of this command looks like::
+
+ - lssap version 1.0 -
+ ==========================================
+   SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+   HA2|  16|       D16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D16/exe
+   HA2|  22|       D22|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D22/exe
+   HA2|  50|       D50|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D50/exe
+   HA2|  51|       D51|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D51/exe
+
+Examples:
+    >>> lssap = shared[Lssap]
+    >>> lssap.instances()
+    ['D16', 'D22', 'D50', 'D51']
+    >>> lssap.version('D51')
+    '749, path 10, changelist 1698137'
+    >>> lssap.is_instance()
+    True
+    >>> lssap.is_hana()
+    False
+    >>> lssap.data[3]
+    {'SID': 'HA2', 'Nr': '51', 'Instance': 'D51', 'SAPLOCALHOST': 'lu0417',
+     'Version': '749, path 10, changelist 1698137', 'DIR_EXECUTABLE': '/usr/sap/HA2/D51/exe'}
+    >>> lssap.data[3]['Instance']
+    'D51'
+"""
+from .. import Parser, parser, parse_table
+from insights.parsers import ParseException
+
+
+@parser('lssap')
+class Lssap(Parser):
+    """Class to parse ``lssap`` command output.
+
+    Attributes:
+        data (list): List of dicts, where the keys in each dict are the column
+            headers and each item in the list represents a SID.
+
+    Raises:
+        ParseException: Raised if any error occurs parsing the content.
+    """
+    instance_dict = {'D': 'netweaver',
+                     'HDB': 'hana',
+                     'ASCS': 'ascs'}
+
+    def __init__(self, *args, **kwargs):
+        self.data = {}
+        super(Lssap, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        # remove lssap version and bar text from content
+        clean_content = content[2:-1]
+        invalid_inst = []
+        if len(clean_content) > 0 and "SID" in clean_content[0]:
+            self.data = parse_table(clean_content, delim='|', header_delim=None)
+        else:
+            raise ParseException("Lssap: Unable to parse {0} line(s) of content: ({1})".format(len(content), content))
+
+        self.running_inst_types = [row["Instance"].rstrip('1234567890') for row in self.data if "Instance" in row]
+
+        for i in self.running_inst_types:
```

Comment:
Just like the list-comprehension used in line-69, the `for` loop at line-71 can be simplified as:

~~~
        invalid_inst = [i for i in self.running_inst_types if i not in self.instance_dict]

~~~
btw. the `.keys()` an be ignored.